### PR TITLE
Plugins CI 

### PR DESCRIPTION
**Pull request by mohfh**
Complete CI at least one step (Build) and make the ci demo more dynamic to changes

### CODE REVIEW

File: .ci/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+RUN chown $_LOGIN $_HOME
+USER root
+
+ENV TZ=Israel
+RUN ln -snf /usr/share/zoneinfo/$TZ /etc/localtime && echo $TZ > /etc/timezone
+
+RUN apt clean && \
+    apt update && \
+    apt upgrade -y && \
+    apt install -y python3 python3-pip apt-transport-https sqlite3 sudo git
```
Comment (lennybe) [inline .ci/Dockerfile:19]: please install all packages in the single RUN
this approach will create less container layers

File: .ci/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+RUN apt-get update && \
+    apt-get install -y \
```
Comment (lennybe) [inline .ci/Dockerfile:22]: why do you need to run  each time? ones is enough.
apt-get update
apt upgrade -y

File: README.md [hunk 3]
```diff
@@ -41,3 +41,33 @@ Install my-project with npm
...
+
+# Adding Plugins to CI Pipeline
+
+This project supports integration of custom plugins into the Continuous Integration (CI) pipeline. The process for integrating your plugin is simple and described below.
+
+## Prerequisites
+
+Your plugin should have a dedicated `.ci` directory that contains a `ci_matrix.yaml` file. This file is used by the CI pipeline to manage the build process for your plugin.
+
+You can use the `ci_matrix.yaml` file found in the `hello_world_plugin` directory as a template.
```
Comment (lennybe) [inline README.md:53]: pls add direct URL to ci_matrix.yaml example

File: plugins/hello_world_plugin/.ci/ci_matrix.yaml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+
+steps:
+  - name: Build Plugin
+    containerSelector: "{name: 'plugin_worker'}"
+    run: |
+      cd plugins/$plugin_dir/build
+      bash -x ./docker_build.sh latest /
+      ls -l /
+      cp /ufm-plugin* /auto/UFM/tmp/${JOB_NAME}/${BUILD_ID}/
```
Comment (lennybe) [inline plugins/hello_world_plugin/.ci/ci_matrix.yaml:32]: not sure that all plugins have this naming convention

Comment (mohfh) [reply plugins/hello_world_plugin/.ci/ci_matrix.yaml:32]: All of them have this prefix 
image

**Review by lennybe — APPROVED** [review]

**Pull request merged**